test/Libraries/Microsoft.AspNetCore.Telemetry.Middleware.Tests/Metering/HttpMeteringTests.cs

```diff
@@ -49,7 +49,7 @@ public HttpMeteringTests()
     [Fact]
     public async Task CanLogIncomingRequestMetric()
     {
-        using var metricCollector = new MetricCollector(new List<string> { typeof(HttpMeteringMiddleware).FullName! });
+        using var metricCollector = new MetricCollector<long>(typeof(HttpMeteringMiddleware).FullName!, null, Metric.IncomingRequestMetricName);
```

Comment:
You probably want to start using IMeterFactory and specifying a scope for these tests. Binding by name in global scope makes the tests non-deterministic when run in parallel with other unit tests manipulating a Meter of the same name.

Reply:
Can you show me an example?

The test is trying to bind to a meter that's embedded in the implementation of the middleware. So do you mean using a meter factory in the middleware itself, or merely in the test?